### ISSUE TITLE
[ASPNET] Generated code does not compile with .Net Core

### ISSUE THREAD

**Issue by frostyandy2k** [swagger-api/swagger-codegen#3337]
##### Description

After writing the yaml for a hello world API with the Swagger online editor, I downloaded the ASPNET5 server and tried to run it using .Net Core (https://www.microsoft.com/net/core). It was run inside the official .Net Core docker container (https://hub.docker.com/r/microsoft/dotnet/).

```
root@b0c0ac1c7942:/webserver# dotnet restore
log  : Restoring packages for /webserver/project.json...
log  : Installing System.Runtime.Handles 4.0.1-beta-23516.
.......
log  : Installing System.Threading.Tasks 4.0.10.
log  : Installing System.Resources.ResourceManager 4.0.1-beta-23516.
log  : Installing System.Linq.Expressions 4.0.11-beta-23516.
error: Package Microsoft.CodeAnalysis.CSharp 1.1.0-rc1-20151109-01 is not compatible with dnxcore50 (DNXCore,Version=v5.0). Package Microsoft.CodeAnalysis.CSharp 1.1.0-rc1-20151109-01 supports:
error:   - net45 (.NETFramework,Version=v4.5)
error:   - portable-net45+win8 (.NETPortable,Version=v0.0,Profile=Profile7)
error: Package Microsoft.CodeAnalysis.Common 1.1.0-rc1-20151109-01 is not compatible with dnxcore50 (DNXCore,Version=v5.0). Package Microsoft.CodeAnalysis.Common 1.1.0-rc1-20151109-01 supports:
error:   - net45 (.NETFramework,Version=v4.5)
error:   - portable-net45+win8 (.NETPortable,Version=v0.0,Profile=Profile7)
error: One or more packages are incompatible with DNXCore,Version=v5.0.
log  : Writing lock file to disk. Path: /webserver/project.lock.json
log  : /webserver/project.json
log  : Restore failed in 15726ms.

Errors in /webserver/project.json
    Package Microsoft.CodeAnalysis.CSharp 1.1.0-rc1-20151109-01 is not compatible with dnxcore50 (DNXCore,Version=v5.0). Package Microsoft.CodeAnalysis.CSharp 1.1.0-rc1-20151109-01 supports:
      - net45 (.NETFramework,Version=v4.5)
      - portable-net45+win8 (.NETPortable,Version=v0.0,Profile=Profile7)
    Package Microsoft.CodeAnalysis.Common 1.1.0-rc1-20151109-01 is not compatible with dnxcore50 (DNXCore,Version=v5.0). Package Microsoft.CodeAnalysis.Common 1.1.0-rc1-20151109-01 supports:
      - net45 (.NETFramework,Version=v4.5)
      - portable-net45+win8 (.NETPortable,Version=v0.0,Profile=Profile7)
    One or more packages are incompatible with DNXCore,Version=v5.0.
```

##### Swagger-codegen version

Using the online swagger editor.
Swagger Editor Version 2.10.1

##### Steps to reproduce

Define an API, download ASPNET5 server.
`> dotnet restore`

##### Fix that worked

The fix from @salerth works:
https://github.com/aspnet/Home/issues/1265

The code generator should add an additional "imports" line to the project.json, so that it includes the missing libraries:
```
 "frameworks": {
        "dnxcore50": { 
            "imports": "portable-net451+win8"
        }
    }
```

**Comment by wing328** [comment]
@frostyandy2k please give it another try as there's been some enhancements to `aspnet5` generator.

**Comment by wing328** [comment]
cc @jimschubert

**Comment by jimschubert** [comment]
I just ran this about a week ago on the 2.2.0 branch and it worked. I can run again, but I have a few questions.

When you say you ran it against the official image... Which version? 

The generator for swagger codegen 2.1.6 used .Net Core 1.0.0-rc1-final assemblies and the official image `microsoft/aspnet:1.0.0-rc1-final`. Version 2.2.0 was updated to preview 2, which has breaking changes over the previous version. It now uses the official image `microsoft/dotnet:latest`.

Another question... Did the code not generate with a Dockerfile for you? See: https://github.com/swagger-api/swagger-codegen/blob/master/samples/server/petstore/aspnet5/src/IO.Swagger/Dockerfile The generator should be ready to go from both a development and deployment perspective. If there's no Dockerfile under src/Project then that's another issue that'll need to be created.

When I tested this last, it ran in Windows 8, OS X, and the container without additional dependencies being added. The point of .Net Core is that you shouldn't have to keep referencing .Net 4.5.x unless you're using dependencies which require it, but that's kind of messed up between rc1 and preview 2.

In your output, it complains about rc1 dependencies but you're running with the newer `dotnet` cli. My assumption is the case with using the correct image as mentioned above, preferably via the generated Dockerfile. I wanted to mention, though, if you're coding on a Mac there's no clean upgrade from rc1 to preview-2. You'll need to fully uninstall rc1 to install preview-2. I'll see if I can track down the issue and link it here.
